src/ForumReader.ts

```diff
@@ -82,13 +82,17 @@ export default class ForumReader {
      * @returns The node with the specified ID
      * @throws {Error} Thrown if an AnswerHubAPI error occurs
      */
-    private async getNode(id: number): Promise<Node> {
+    private async getNode(id: number): Promise<Node | undefined> {
         if (this.cachedNodes.has(id)) {
             return this.cachedNodes.get(id)!;
         } else {
-            const node = await this.answerHub.getNode(id);
-            this.cachedNodes.set(id, node);
-            return node;
+            try {
+                const node = await this.answerHub.getNode(id);
+                this.cachedNodes.set(id, node);
+                return node;
+            } catch (error) {
+                console.error(`Error occurred while making a request to the answerhub api: ${error}`);
```

Comment:
So while this is a valid way to indicate errors, it's a bit "unconventional". I'm fine with both options (resolving as undefined or rejecting), but be sure to update the documentation since right now its incorrect.